src/category_theory/limits/shapes/products.lean

```diff
@@ -0,0 +1,49 @@
+-- Copyright (c) 2018 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Scott Morrison
+
+import category_theory.limits.limits
+import category_theory.discrete_category
+
+open category_theory
+
+namespace category_theory.limits
+
+universes v u
+
+variables {β : Type v}
+variables {C : Sort u} [𝒞 : category.{v+1} C]
+include 𝒞
+
+-- We don't need an analogue of `pair` (for binary products), `parallel_pair` (for equalizers),
+-- or `(co)span`, since we already have `functor.of_function`.
```

Comment:
In the other files you have specialised versions of the cone projections etc. Does it make sense to also have them for this file? Or does the generic version work just as well?